src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -16,12 +17,14 @@
    * Type matches the type of the entities generated by the Natural Language API.
    * All types can be seen here: 
    * https://cloud.google.com/natural-language/docs/reference/rest/v1/Entity#type
+   * 
+   * Each entity will have multiple, non-duplicate types.
    */ 
-  private final String type;
+  private HashSet<String> type;
```

Comment:
We only care about the interface type here. `HashSet` is not the only implementation of `Set` (other examples are `LinkedHashSet` and `TreeSet`), and we should use `Set` instead. This helps to achieve polymorphism and in general might help you when writing tests. For example, your tests could have your own implementation of the `Set` interface, and you could have more control over it. This might not be obvious now, but when you implement more complex structures in the future you'll definitely find it useful to have a production vs test implementation of an interface.